### PR TITLE
Add Optional Postgres SSL Connection Parameters

### LINKED ISSUE #3344: Optional SSL parameters to the Postgres connector
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/reference/warehouse-setups/postgres-setup#profile-configuration

### What part(s) of the page would you like to see updated?

https://github.com/dbt-labs/dbt-core/pull/3473 added three new config options as seen [here](https://github.com/dbt-labs/dbt-core/blob/810ef7f5560b4bbe8a3afb0dc182a4380d3a3dcd/plugins/postgres/dbt/adapters/postgres/connections.py#L31-L33):
- `sslcert`
- `sslkey`
- `sslrootcert`

But they aren't documented, so they aren't very searchable (outside of the source code).

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by jefflinwood**
## What are you changing in this pull request and why?

Adding three parameters for Postgres client SSL connections that were previously undocumented.

Fixes https://github.com/dbt-labs/docs.getdbt.com/issues/3344

## Checklist

- [X] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.

### CODE REVIEW

File: website/docs/reference/warehouse-setups/postgres-setup.md
Comment (nghi-ly) [inline website/docs/reference/warehouse-setups/postgres-setup.md]: sugg using the exact title of the section and calling it out that it's in the postgresql docs (for clarity in case people might assume it's in the dbt docs)

```suggestion
The `sslcert` config controls the location of the certificate file used to connect to Postgres when using client SSL connections. To use a certificate file that is not in the default location, set that file path using this value. Without this config set, dbt uses the Postgres default locations. See [Client Certificates](https://www.postgresql.org/docs/current/libpq-ssl.html#LIBPQ-SSL-CLIENTCERT) in the PostgreSQL SSL docs for the default paths.
```

File: website/docs/reference/warehouse-setups/postgres-setup.md
Comment (nghi-ly) [inline website/docs/reference/warehouse-setups/postgres-setup.md]: similar sugg as previvous

**Review by nghi-ly — COMMENTED** [review]
Thank you so much, @jefflinwood , for addressing issue 3344! Really appreciate the contribution! Left a couple suggs.

Love meeting you and our great chat during WTD Writing Day! :)

**Review by nghi-ly — COMMENTED** [review]
Thank you so much, @jefflinwood , for addressing #3344! Really appreciate the contribution! Left a couple suggs.

Love meeting you and our great chat during WTD Writing Day! :)

File: website/docs/reference/warehouse-setups/postgres-setup.md
Comment (jefflinwood) [reply website/docs/reference/warehouse-setups/postgres-setup.md]: Great suggestion!

File: website/docs/reference/warehouse-setups/postgres-setup.md
Comment (jefflinwood) [reply website/docs/reference/warehouse-setups/postgres-setup.md]: Updated, but would appreciate a review too!

**Review by dataders — DISMISSED** [review]
very helpful additions -- thanks! can you also add this parameters to the example file above? I'm not sure how to lever these params within a profile.

https://github.com/dbt-labs/docs.getdbt.com/blob/a57b90ce84a1158e70d8adb8cfafebc09ea739ac/website/docs/reference/warehouse-setups/postgres-setup.md?plain=1#L74

Relatedly, in #429, @dbeatty10 documented (https://github.com/dbt-labs/dbt-redshift/issues/429#issuecomment-1536777007) more on the SSL parameter options that are exposed within dbt-redshift. might be work taking a look. 


👀 @mirnawong1 because we've talked about how to best document this "optional parameters"

**Comment by dbeatty10** [comment]
Thanks for jumping into this @jefflinwood ❤️ !

Could you update `Fixes issue https://github.com/dbt-labs/docs.getdbt.com/issues/3344` in the PR description to be this instead?
```
Fixes https://github.com/dbt-labs/docs.getdbt.com/issues/3344
```
With the latter, GitHub will automatically link the PR with the issue and auto-close the issue once the PR is merged. With the current syntax, the issue will need to be closed manually.

**Comment by mirnawong1** [comment]
hey @dataders, thanks for linking me to that table, will keep it in mind! 

also I've added the new parameters to the example file - can you give it a glance to make sure it looks technically right?

**Comment by dbeatty10** [comment]
@mirnawong1 all looks good to me from a technical standpoint 👍

**Comment by mirnawong1** [comment]
woohoo! thank you so much @dbeatty10 !

**Comment by mirnawong1** [comment]
hey @dataders going to approve this and merge unless you (or @dbeatty10 ) have other feedback!

**Comment by mirnawong1** [comment]
Approving and merging this now

**Review by mirnawong1 — APPROVED** [review]

**Review by mirnawong1 — APPROVED** [review]

**Pull request merged**
